### PR TITLE
README: add CI build badges

### LINKED ISSUE #35: Add CI build badges
Blocked by https://github.com/libusb/hidapi/pull/34

Badges look like this: https://builds.sr.ht/~qbicz/hidapi-1/archlinux.yml

### PR DESCRIPTION
**Pull request by Qbicz**
Show status of test builds on the project GitHub page.
Fixes #35.

Signed-off-by: Filip Kubicz <filip@kubicz.engineer>

### CODE REVIEW

**Comment by Qbicz** [comment]
Preview available here: https://github.com/Qbicz/hidapi/tree/ci_build_badges

@Youw @z3ntu @todbot can you review?

**Comment by z3ntu** [comment]
LGTM

**Comment by Youw** [comment]
I'd add explicit names to those badges, e.g.:

```
macOS master build: <badge> Last Linux/BSD build: <badge>
```
At least because builds.sr.ht shows any last, maybe some PR (possibly failing) build.

**Comment by Qbicz** [comment]
I added the explicit names. First in single line but it looked ugly, now in separate lines. But still it looks horrible. I would prefer badges only ;)

But actually a badge sends you to a dashboard where you can check what went wrong, and any failure should raise an alert for us. So having just [travis badge] [sourcehut badge] looks ok for me. It does not create too much noise on the readme page.

**Comment by Youw** [comment]
What I'm trying to fix here - is a small user experience:

Imagine someone opens a repository, and first what gets to see - a broken build badge (more than possible, because, maybe, someone submitted a broken PR).
Most people wouldn't go and check which build is broken, and that master is actually just fine, but rather just think that "oh, master here is broken".
Maybe this someone even don't need master, but it just leaves bad first impression. Personally I don't like bad first impressions)

Second story: you're a very curious user, and want to see the build logs for yourself, but only on macOS, and you know, that one of the badges (likely) is from macOS CI. Which one do you need to open? Try all of them? Just not convenient.

Regarding ugliness - maybe it really doesn't look very nice... What about having all of them in a table grid?

**Comment by Qbicz** [comment]
@Youw I understand. Added the badges with titles to the table grid.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## HIDAPI library for Windows, Linux, FreeBSD and macOS
 
+| Build type           | Status |
+|----------------------|--------|
+| `macOS master`       | [![Build Status](https://travis-ci.org/libusb/hidapi.svg?branch=master)](https://travis-ci.org/libusb/hidapi) |
+| `Linux/BSD, last PR` | [![builds.sr.ht status](https://builds.sr.ht/~qbicz/hidapi.svg)](https://builds.sr.ht/~qbicz/hidapi?) |
```
Comment (Youw) [inline README.md]: > last PR

this is not very accurate

"last build (branch/PR)" - accurate, but too verbose (and long)

maybe some other suggestion?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 ## HIDAPI library for Windows, Linux, FreeBSD and macOS
 
+| Build type           | Status |
```
Comment (Youw) [inline README.md]: Doesn't seems like a "type" to me. More like a "Build group" or "CI instance" (since one CI instance may include several platforms)

Comment (Qbicz) [reply README.md]: Sticked to CI instance as it describes best what's happening.

File: README.md (hunk 1 above)
Comment (Qbicz) [reply README.md]: The accurate description looks best here, thanks!

**Comment by Qbicz** [comment]
@Youw added Windows Appveyor status and introduced your changes. Does it look ok now?

**Comment by Qbicz** [comment]
@z3ntu do you have any idea why FreeBSD build cannot find autoconf 2.69? https://builds.sr.ht/~qbicz/job/77500

**Comment by z3ntu** [comment]
I saw that, I quote from the #sr.ht IRC:
> wut
> \> FreeBSD repository is up to date.
> this one is probably some sort of race condition, since autoconf-2.69_3.txz is there

**Comment by Qbicz** [comment]
This really looks like a race, since after resubmitting the build it passed.

**Pull request merged**